getsmarter_api_clients/geag.py

```diff
@@ -124,3 +124,105 @@ def create_allocation(
 
         response = self.post(url, json=payload)
         response.raise_for_status()
+
+    # This is for the newer endpoint created by GetSmarter for enterprise
+    # specific needs.
+    def create_enterprise_allocation(
+        self,
+        payment_reference,
+        first_name,
+        last_name,
+        email,
+        date_of_birth,
+        terms_accepted_at,
+        currency,
+        order_items,
+        address_line1=None,
+        address_line2=None,
+        city=None,
+        postal_code=None,
+        state=None,
+        state_code=None,
+        country=None,
+        country_code=None,
+        mobile_phone=None,
+        work_experience=None,
+        education_highest_level=None
+    ):
+        """
+        Create an enterprise_allocation (enrollment) through GEAG.
+
+        :Parameters:
+          - `payment_reference (str)`: Reference used by enterprise partner
+            when payment is made to GetSmarter
+          - `first_name (str)`: First name
+          - `last_name (str)`: Last name
+          - `email (str)`: Email
+          - `date_of_birth (str)`: Date of birth
+          - `terms_accepted_at (str)`: ISO 8601 timestamp of
+            when the terms and policies were accepted
+          - `currency (str)`: One of ['USD', 'GBP', 'ZAR', 'EUR', 'AED',
+            'SGD', 'HKD', 'SAR', 'INR', 'CAD']
+          - `order_items (list of dict)`: Items ordered
+          - `address_line1 (str)`: Address Line 1
+          - `address_line2 (str)`: Adress Line 2
+          - `city (str)`: City
+          - `postal_code (str)`: Postal code
+          - `state (str)`: State
+          - `state_code (str)`: State code
+          - `country (str)`: Country
+          - `country_code (str)`: Country code
+          - `mobile_phone (str)`: Mobile phone number
+          - `work_experience (str)`: One of ['None', '1 to 5 years',
+            '5 to 15 years', 'More than 15 years']
+          - `education_highest_level (str)`: One of ['High school',
+            'Bachelor’s degree', 'Master’s degree', 'Doctoral degree',
+            'Other tertiary qualification', 'Honours degree',
+            'Bachelors degree']
+
+        **Example payload**
+          { "paymentReference": "GS-12304",
+            "firstName": "Jan",
+            "lastName": "Pan",
+            "email": "janpan@gs.com",
+            "dateOfBirth": "2021-05-12",
+            "termsAcceptedAt": "2021-05-21T17:32:28Z",
+            "currency": "ZAR",
+            "orderItems": [{ "productId": "product_id", "quantity": 1,
+            "normalPrice": 1000, "discount": 500, "finalPrice": 500 }],
+            "addressLine1": "Oak Glen",
+            "city": "Cape Town",
+            "postalCode": "7570",
+            "country": "South Africa",
+            "countryCode": "ZA" }
+
+        """
+        url = f'{self.api_url}/enterprise_allocations'
```

Comment:
This looks like the only line that's different from `create_allocation()` above.  Could we DRY this up?

Reply:
I didn't want to DRY anything up yet because I believe there are changes TBD coming to create_enterprise_allocation, but it wasn't clear to me which (because there are no swagger docs for them yet). But, noted. When we see what the new endpoint needs this can be collapsed a bit